packages/shared-ux/error_boundary/src/ui/message_components.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+
+import {
+  EuiButton,
+  EuiCodeBlock,
+  EuiEmptyPrompt,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiLink,
+  EuiTitle,
+  useGeneratedHtmlId,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiCopy,
+} from '@elastic/eui';
+
+import { errorMessageStrings as strings } from './message_strings';
+
+export interface ErrorCalloutProps {
+  error: Error;
+  errorInfo: Partial<React.ErrorInfo> | null;
+  name: string | null;
+  reloadWindow: () => void;
+}
+
+const CodePanel = (props: ErrorCalloutProps & { onClose: () => void }) => {
```

Comment:
nit: your code is perfectly good, just a tiny improvement could be to explicitly say that this is not just a function, but a React component:

```suggestion
const CodePanel: Ract.FC<ErrorCalloutProps & { onClose: () => void }> = (props) => {
```